### ISSUE TITLE
CPU memory in use increases during Training

### ISSUE THREAD

**Issue by tetsu-kikuchi** [google/automl#224]
I followed your instruction "6. Train on PASCAL VOC 2012 with backbone ImageNet ckpt" and "7. Finetune on PASCAL VOC 2012 with detector COCO ckpt".  
In both cases, I encountered a problem that CPU memory in use increases during training.  
The following is the result of free --giga -s 1800, during the training: 
``` 
             total        used        free      shared  buff/cache   available
Mem:             65           1          19           0          44          63
Swap:             2           0           2

              total        used        free      shared  buff/cache   available
Mem:             65           6          13           0          44          58
Swap:             2           0           2

              total        used        free      shared  buff/cache   available
Mem:             65           8          11           0          44          55
Swap:             2           0           2

              total        used        free      shared  buff/cache   available
Mem:             65          10           9           0          45          54
Swap:             2           0           2
...
...
```
And the process was eventually killed when the CPU memory in use reached its maximum 
(after about 18 hours for D0;  the time to be killed is shorter when I used D2, about 12 hours).
My OS is Ubuntu 18.04.

I couldn't find the reason by myself.  I am afraid that this problem might affect the overall training time, and would appreciate if you could modify this matter in future revision.  Thanks.

**Comment by dhiaul98** [comment]
i have same problem with you

**Comment by omair50** [comment]
same issue

**Comment by xueweili** [comment]
Did you solve this problem? I have same issue.

**Comment by tetsu-kikuchi** [comment]
> Did you solve this problem? I have same issue.

As for me, unfortunately not.

**Comment by shreymohan** [comment]
Did anyone solve this issue? @mingxingtan @fsx950223 any suggestions?

**Comment by kahkeng** [comment]
I suspect there's some memory leak in or after the eval phase that runs after each epoch of training, which doesn't get freed and accumulates as the number of epochs go up. In one run, I got through only 3 epochs (~40K images) before training ended. I modified num_examples_per_epoch to be 10x larger and managed to get 10x more training done with the same number of eval passes. Something to consider if you don't need eval to run after every epoch.
